src/pages/index.js

```diff
@@ -37,61 +60,106 @@ profileEditButton.addEventListener("click", () => {
 };
 
-function handleAddCardSubmit(data) {
-  renderCard(data);
-  addCardPopup.close();
-  addFormValidator.toggleButtonState();
-}
-
-function handleEditProfileSubmit(data) {
-  userInfo.setUserInfo(data);
-  editProfilePopup.close();
-  editFormValidator.toggleButtonState();
-}
+const addCardPopup = new PopupWithForm(addCardModal, (data) => {
+  addCardPopup.saving(true);
+  api.addCard(data).then((data) => {
+    addCardPopup.saving(false);
+    renderCard(data);
+    addCardPopup.close();
+    addFormValidator.toggleButtonState();
+  });
+});
+addCardPopup.setEventListeners();
 
 function handleImageClick(name, link) {
   previewImagePopup.open(name, link);
 }
 
-const cardSection = new Section(
-  {
-    renderer: renderCard,
-    items: initialCards,
-  },
-  selectors.cardSection
-);
+const previewImagePopup = new PopupWithImage(imageModal, handleImageClick);
+previewImagePopup.setEventListeners();
 
-cardSection.renderItems();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api.removeLike(card._id).then(() => {
+      card.handleLikeButton();
+    });
+  } else {
+    api.likeCard(card._id).then(() => {
+      card.handleLikeButton();
+    });
+  }
+}
 
-const addCardPopup = new PopupWithForm(addCardModal, handleAddCardSubmit);
-addCardPopup.setEventListeners();
+const deleteCardPopup = new PopupWithConfirm(deleteCardModal);
+deleteCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  profileEditModal,
-  handleEditProfileSubmit
-);
-editProfilePopup.setEventListeners();
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.deleting(true);
+    api.deleteCard(card._id).then(() => {
+      deleteCardPopup.deleting(false);
+      card.removeCard();
+      deleteCardPopup.close();
+    });
+  });
+}
 
-const previewImagePopup = new PopupWithImage(imageModal, handleImageClick);
-previewImagePopup.setEventListeners();
+avatarEditButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-const userInfo = new UserInfo(profileTitle, profileDescription);
+const editAvatarPopup = new PopupWithForm(avatarEditModal, (data) => {
+  editAvatarPopup.saving(true);
+  api.setUserAvatar(data).then((data) => {
+    editAvatarPopup.saving(false);
+    userInfo.setAvatar(data.avatar);
+    editAvatarPopup.close();
+    avatarFormValidator.toggleButtonState();
+  });
+});
+editAvatarPopup.setEventListeners();
 
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const addFormValidator = new FormValidator(validationSettings, addCardForm);
+
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
+
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 editFormValidator.toggleButtonState();
 addFormValidator.toggleButtonState();
+avatarFormValidator.toggleButtonState();
```

Comment:
You can call `toggleButtonState` inside `FormValidator` only once in method `_setEventListeners` not to do that 3 times in `index.js` here